apps/wallet/src/ui/app/hooks/useGetValidators.ts

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { isSuiObject, isSuiMoveObject } from '@mysten/sui.js';
+import BigNumber from 'bignumber.js';
+import { useMemo } from 'react';
+
+import { useGetObject } from './useGetObject';
+import {
+    STATE_OBJECT,
+    VALDIATOR_NAME,
+} from '_app/staking/usePendingDelegation';
+
+//TODO: Remove when available on SDK, types should come from the SDK
+import type { ValidatorState } from '_app/staking/home/ValidatorDataTypes';
+
+function getStakePercent(stake: bigint, total: bigint): number {
+    const bnStake = new BigNumber(stake.toString());
+    const bnTotal = new BigNumber(total.toString());
+    return bnStake.div(bnTotal).multipliedBy(100).toNumber();
+}
+
+export function useGetValidators(walletAddress: string | null) {
+    const { data, isLoading, isError } = useGetObject(STATE_OBJECT);
+
+    const validatorsData =
+        data && isSuiObject(data.details) && isSuiMoveObject(data.details.data)
+            ? (data.details.data.fields as ValidatorState)
+            : null;
+
+    const totalStake =
+        validatorsData?.validators.fields.total_validator_stake || 0n;
+
+    const validators = useMemo(() => {
+        if (!validatorsData) return [];
+        return validatorsData.validators.fields.active_validators.map((av) => {
+            const rawName = av.fields.metadata.fields.name;
+
+            let name: string;
+
+            if (Array.isArray(rawName)) {
+                name = String.fromCharCode(...rawName);
+            } else {
+                name = Buffer.from(rawName, 'base64').toString();
+                if (!VALDIATOR_NAME.test(name)) {
+                    name = rawName;
+                }
+            }
+            const {
+                sui_balance,
+                starting_epoch,
+                pending_delegations,
+                delegation_token_supply,
+            } = av.fields.delegation_staking_pool.fields;
+
+            const num_epochs_participated =
+                validatorsData.epoch - starting_epoch;
+            const APY =
```

Comment:
Did you verify this formula? I assumed that the `^` in the pseudocode was exponent, but in JS this is bitwise XOR so wanted to make sure this is correct.

Reply:
You are right. totally missed that.

Reply:
done